### PR TITLE
Activate dependabot, build tweaks, green CI

### LINKED ISSUE #11: cleanup: add basics for testing of repos
- liniting
- code clean up 
- tests are failing

@ssallam to review and merge

### PR DESCRIPTION
**Pull request by kremalicious**
- activate dependabot
- move auto-generated typings into `src/@types`
- run typings generation on CI
- run build on CI
- add `src/@types/` & `build/` to `gitignore`
- package updates
- switch to GitHub Actions for CI run
- disable [`eqeqeq`](https://eslint.org/docs/rules/eqeqeq) ESLint rule, because The Graph relies on the non-best practice usage of it
- simplify ESLint defaults from `extends`

Actually making use of type checking stays broken since https://github.com/graphprotocol/graph-ts/issues/113 seems still not solved

closes #11

### CODE REVIEW

File: src/helpers.ts [hunk 1]
```diff
@@ -310,10 +310,10 @@ export function createPoolTransaction(
...
   pool.consumePrice = poolTx.consumePrice
   pool.spotPrice = poolTx.spotPrice
-  const lockedValue = pool.lockedValue
-  pool.lockedValue = pool.oceanReserve + (pool.datatokenReserve * pool.spotPrice)
-  let factory = PoolFactory.load('1')
-  factory.totalLockedValue = factory.totalLockedValue - lockedValue + pool.lockedValue
+  pool.lockedValue = pool.oceanReserve + pool.datatokenReserve * pool.spotPrice
+  const factory = PoolFactory.load('1')
+  factory.totalLockedValue =
+    factory.totalLockedValue - pool.lockedValue + pool.lockedValue
```
Comment (kremalicious) [inline src/helpers.ts:316]: @ssallam did not further touch this but no clue why we are subtracting first and then adding same value again?

Comment (ssallam) [reply src/helpers.ts:316]: in main branch it is:

Comment (ssallam) [reply src/helpers.ts:316]: maybe that was a typo for some short time

Comment (ssallam) [reply src/helpers.ts:316]: it should be adding the new value, and subtracting the old value.

Comment (kremalicious) [reply src/helpers.ts]: perfect thanks! Made the intention more clear in code in 6b304e0

**Comment by mihaisc** [comment]
If no one has any comments i'll merge this today/tomorrow

**Review by mihaisc — APPROVED** [review]

**Comment by kremalicious** [comment]
giphy-2

**Comment by ssallam** [comment]
We can't merge before testing deployment on a local node.

**Comment by alexcos20** [comment]
I agree, as long as there are changes in the actual code (src/).   We need to test it first

**Comment by mihaisc** [comment]
Will test on rinkeby

**Review by ssallam — APPROVED** [review]
LGTM.

**Comment by ssallam** [comment]
Tested on Rinkeby (using local graph-node), seems to work.

**Pull request merged**